Report/Chapters/Design.tex

```diff
@@ -1,17 +1,103 @@
 \chapter{Design}
 
+This chapter contains a high level description of the design choices made in the process of creating Viskell.
+The specific implementation details are discussed in chapter \ref{chap:implementation} on page \pageref{chap:implementation}.
+
+\section{User interface}
+
+The user interface basically consists of two parts, namely the menus for the different options a user can select and the part where the user can manipulate functions and create their own (the canvas).
+
+\subsection{Graphical design}
+
+\subsubsection{Blocks and lines}
+
+We chose to represent Haskell functions by function blocks since this is a logical element for users, a combination of blocks forms a program.
+Each block consists of a function name and its associated arguments. With each argument comes an anchor node to which other block can be connected through lines.
+
+In Haskell, values are being passed to functions without being saved to memory.
+The lines between blocks represent this flow.
+Thus, a program can be built like an electronic circuit.
+
+We chose to make a distinction between three kinds of blocks.
+The first kind in an input block, which represents a value.
```

Comment:
It's called a Value Block in the menu.

You could underline the fact that it is a constant value here, i.e. you would use one where you would use a constant literal in plain ol' Haskell.